suite-native/device/src/hooks/useDeviceAuthenticityCheck.ts

```diff
@@ -65,33 +64,17 @@ export const useDeviceAuthenticityCheck = () => {
         [],
     );
 
-    const createStoredResult = useCallback(
-        (payload: CheckDeviceAuthenticityThunkResult) => {
-            if (
-                payload?.error === 'CA_PUBKEY_NOT_FOUND' &&
-                'configExpired' in payload &&
-                payload?.configExpired
-            ) {
-                // CA_PUBKEY_NOT_FOUND with configExpired is temporarily allowed and just logged to Sentry
-                reportCheckResult('configExpired', payload.error, payload);
-
-                return {
-                    ...payload,
-                    valid: true,
-                };
-            }
+    const createStoredResult = useCallback((result: RawResult): StoredAuthenticateDeviceResult => {
+        // error from the TrezorConnect call itself, always considered a failure.
+        // One of the cases may be isDeviceBootloaderUnlocked, but they are handled equally
+        if (!result.success) {
+            return { valid: false, ...result.payload };
```

Comment:
Are we sure, that `result.payload` is never `undefined`? It looks so from the type, but I'm not so sure.